### ISSUE TITLE
Add Basilisp support to clojure-lsp

### ISSUE THREAD

**Issue by ikappaki** [clojure-lsp/clojure-lsp#1883]
**Is your feature request related to a problem? Please describe.**

I would like to add support in `clojure-lsp` for `Basilisp`, a Clojure-compatible language running on the Python VM.

https://github.com/basilisp-lang/basilisp

https://basilisp.readthedocs.io/en/latest/

Basilisp aims to be compatible with Clojure in terms of s-expressions and libraries but includes some unique features for Python VM interop.

To name a few ways that differs from Clojure

https://basilisp.readthedocs.io/en/latest/differencesfromclojure.html

- Uses the `.lpy` file extension
- Has its own `:lpy` reader conditional
- Supports keyword arguments (python form `name=value`), that can be passed to a Basilisp function after `**` as `:name` value`
- Uses `import` to include Python libraries

Despite these differences, its reader, s-expressions, forms, structure, and libraries are largely compatible with Clojure.

https://basilisp.readthedocs.io/en/latest/reference.html

**Describe the solution you'd like**

I would like clojure-lsp to support `Basilisp` projects similarly to how it supports `Clojure` and `ClojureScript`. What steps should I take to get started, and what considerations should I keep in mind?

I expect that most of the functionality in `clojure-lsp` would already apply to `Basilisp`, but I would need to address the specific gaps related to Python interop features unique to `Basilisp`.

Thanks

**Comment by NoahTheDuke** [comment]
I expect the first step would be support in clj-kondo.

**Comment by ericdallo** [comment]
Yeah, First we need to know what tool will understand the language AST, like what is each token etc, today we rely on clj-kondo for all clojure/cljs/cljd related stuff, not sure would make sense add there but this is the start
